### DISCUSSION TITLE
GLPI Agent - Software deployment

### DISCUSSION THREAD

**Discussion by benmahria** [glpi-project/glpi-agent#467] [Q&A]
### Documentation & bug reporting acknowledgment

Yes, I read it, Not applicable

### Describe your problem

I am using the GLPI project for IT inventory and software deployment. However, there is an issue where when I create a software deployment task, it doesn't execute immediately. Instead, it waits until the server contacts the GLPI agent. If I need the software to be installed immediately, I have to manually force the execution from the browser. Is there a way to resolve this?

**Comment by g-bougard** [comment]
Hi @benmahria 

depending on your network, you may force an agent to run its tasks from GLPI. For a given computer you have a link to force tasks run on its asset page. First check GLPI can get the agent status from that page.

By default, the Deploy task is run when the agent contacts GLPI server.

**Reply by kintaro1981** [reply]
the link to force tasks on its asset page is simply the "request inventory"? or is there another link to force only tasks?

**Reply by g-bougard** [reply]
Hi @kintaro1981 

yes, this indeed asks agent to run all its tasks, including inventory, deploy, etc...

Of course, this can only work if glpi has a direct access to 62354 glpi-agent port (but you may have configured another port), without firewall filtering.

**Comment by kintaro1981** [comment]
- On demand packages can be deployed immediatly, you can launch them from the Deploy Package tab in the Computer page.
- Packages inside active tasks (tasks can include multiple deploy packages) starts when clients starts inventory manually or scheduled.
N.B. Task need to be active, you can set a start date in which glpi prepare the jobs for the actors or force start.
